tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,231 @@
+    tags: Optional[List[str]] = None,
+    dataset: Optional[Dataset] = None,
+) -> Dataset:
+    """Flexibly create a source dataset in Tamr. Will use array string as default attribute type
+       if none are specified. If a dataset object is passed, a new dataset with dataset_name as
+       its name will be created that has the same attributes and primary keys as the dataset.
+
+    Args:
+        tamr: TUC client
+        dataset_name: name for new dataset
+        primary_keys: one or more attributes for primary key(s) of new dataset
+        attributes: list of attribute names for new dataset
+        attribute_types: dict of attribute types, attribute name is key and type is value
+        description: text description of new dataset
+        external_id: external_id for dataset, if None Tamr will create one for you
+        tags: tags for new dataset
+        dataset: optional dataset TUC object that the new dataset will replicate
+
+    Returns:
+        Dataset created in Tamr
+
+    Raises:
+        requests.HTTPError: If any HTTP error is encountered
+        ValueError: Dataset or primary_keys must be defined
+        ValueError: A dataset with name '{dataset_name}' already exists
+    """
+
+    if not dataset and not primary_keys:
+        raise ValueError(f"Dataset or primary_keys must be defined")
+
+    # Get dataset information
+    if dataset:
+        # Get attributes from dataset object
+        attribute_specs = [attr.spec() for attr in dataset.attributes.stream()]
+        # Get dataset spec information
+        description = dataset.description
+        tags = dataset.tags
+        primary_keys = dataset.key_attribute_names
+    else:
+        attribute_specs = _create_specs(
+            attribute_names=attributes, attribute_types=attribute_types
+        )
+
+    dataset_exists = exists(client=client, dataset=dataset_name)
+    if not dataset_exists:
+        creation_spec = {
+            "name": dataset_name,
+            "description": description,
+            "keyAttributeNames": primary_keys,
+            "externalId": external_id,
+            "tags": tags,
+        }
+        client.datasets.create(creation_spec)
+    else:
+        raise ValueError(f"A dataset with name '{dataset_name}' already exists")
+
+    # Get new dataset
+    target_dataset = client.datasets.by_name(dataset_name)
+
+    # Get current dataset attributes
+    target_dataset_attributes = target_dataset.attributes
+
+    # Update attributes in dataset
+    for attribute in attribute_specs:
+        attr_spec_dict = attribute.to_dict()
+        attribute_name = attr_spec_dict["name"]
+        if attribute_name in primary_keys:
+            # This attribute already exists
+            continue
+        else:
+            # Create new attribute
+            target_dataset_attributes.create(attr_spec_dict)
+
+    return target_dataset
+
+
+def update(
+    *,
+    dataset: Dataset,
+    primary_keys: List[str],
```

Comment:
I think lets make them optional, it'll be more flexible
```suggestion
    primary_keys: Optional[List[str]] = None,
```

Reply:
Currently primary_keys are no longer an arg for update